File: 백준/Bronze/1236. 성 지키기/성 지키기.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int N = scan.nextInt();
        int M = scan.nextInt();

        boolean[] row = new boolean[M];
        boolean[] column = new boolean[N];

        int rowCount = 0;
        int columnCount = 0;

        for (int i = 0; i < N; i++) {
            char[] line = scan.next().toCharArray();
            for (int j = 0; j < M; j++) {
                if (line[j] == 'X') {
                    if (!column[i]) {
                        column[i] = true;
                        columnCount++;
                    }
                    if (!row[j]) {
                        row[j] = true;
                        rowCount++;
                    }
                }
            }
        }

        System.out.print(Math.max(M - rowCount, N - columnCount));
    }
}

File: 백준/Bronze/1236. 성 지키기/성 지키기.py
N, M = map(int, input().split(" "))
arr = [list(input()) for _ in range(N)]

row, column = [False] * M, [False] * N
rCount, cCount = M, N

for i in range(N):
    for j in range(M):
        if arr[i][j] == "X":
            if not row[j]:
                row[j] = True
                rCount -= 1
            if not column[i]:
                column[i] = True
                cCount -= 1

print(max(rCount, cCount))
